components/AdditionalDatasheetEditor.tsx

```diff
@@ -0,0 +1,295 @@
+'use client';
+
+import { useCallback, useEffect, useState } from 'react';
+import { useQuery, useMutation } from '@apollo/client';
+import {
+  Accordion as MuiAccordion,
+  AccordionDetails as MuiAccordionDetails,
+  AccordionSummary as MuiAccordionSummary,
+  Box,
+  Typography,
+  SxProps,
+  Theme,
+  Skeleton,
+} from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { DataGrid, GridColDef } from '@mui/x-data-grid';
+import { ChevronDown } from 'react-bootstrap-icons';
+
+import { GET_MEASURE_TEMPLATES } from '@/queries/get-measure-templates';
+import { UPDATE_MEASURE_DATAPOINT } from '@/queries/update-measure-datapoint';
+import { useFrameworkInstanceStore } from '@/store/selected-framework-instance';
+import { useSnackbar } from './SnackbarProvider';
+import { additionalMeasures } from '@/constants/measure-overrides';
+import CustomEditComponent from './DatasheetEditor';
+import { CustomFooter } from './DatasheetEditor';
+
+const Accordion = styled(MuiAccordion)(({ theme }) => ({
+  '&:first-of-type': {
+    borderTopLeftRadius: theme.shape.borderRadius,
+    borderTopRightRadius: theme.shape.borderRadius,
+  },
+  '&:last-of-type': {
+    borderBottomLeftRadius: theme.shape.borderRadius,
+    borderBottomRightRadius: theme.shape.borderRadius,
+  },
+  border: `1px solid ${theme.palette.divider}`,
+  '&:not(:last-child)': {
+    borderBottom: 0,
+  },
+  '&::before': {
+    display: 'none',
+  },
+}));
+
+const DATA_GRID_SX: SxProps<Theme> = (theme) => ({
+  '& .MuiDataGrid-columnHeaderTitle, & .MuiDataGrid-cell': {
+    display: 'flex',
+    fontSize: theme.typography.caption.fontSize,
+    whiteSpace: 'normal',
+    lineHeight: 'normal',
+    py: 1,
+    justifyContent: 'flex-start',
+    textAlign: 'left',
+  },
+  '& .MuiDataGrid-cell': {
+    textAlign: 'left',
+  },
+});
+
+interface KeyMeasure {
+  label: string;
+  question: string;
+}
+
+interface MeasureDataPoint {
+  id: string;
+  question: string;
+  baselineYear: string | number;
+  unit: string;
+  [year: number]: string | number;
+}
+
+type MeasureSection = Record<string, MeasureDataPoint[]>;
+
+export function AdditionalDatasheetEditor() {
+  const baselineYear =
+    useFrameworkInstanceStore((state) => state.baselineYear) ?? 0;
+  const selectedInstanceId = useFrameworkInstanceStore(
+    (state) => state.selectedInstance
+  );
+
+  const { setNotification } = useSnackbar();
+  const [measureSection, setMeasureSection] = useState<MeasureSection>({});
+  const [expanded, setExpanded] = useState<number | null>(0);
+
+  const { data, loading, error } = useQuery(GET_MEASURE_TEMPLATES, {
+    variables: { frameworkConfigId: selectedInstanceId },
+  });
+
+  const currentYear = new Date().getFullYear();
+  const additionalYears = Array.from(
+    { length: currentYear - baselineYear - 1 },
+    (_, i) => baselineYear + 1 + i
+  );
```

Comment:
This will be one of the issues causing the infinite loop rerender as `Array.from` creates a new array on each render. The `useEffect` on line 156 will check whether the dependencies have changed each render, and since we have a brand new array on every render which is passed to the dependencies array, it'll call the `useEffect` function every time.

We can fix this by wrapping `additionalYears` in a `useMemo`, which will only create a new array when the `baselineYear` changes (we could also move `currentYear` outside of the component to make it clear to React that it won't change between renders which would allow us to remove it from the `useMemo` dependency array).

```suggestion
  const additionalYears = useMemo(
    () =>
      Array.from(
        { length: currentYear - baselineYear - 1 },
        (_, i) => baselineYear + 1 + i
      ),
    [currentYear, baselineYear]
  );
```

I expect there are a few of these issues so we might need more `useMemo` and `useCallback` hooks in this function. React becomes pretty complicated when there are a lot of dynamic arrays/ objects which also determine whether other things should render. Ping me on Slack if you'd like to huddle on this!